src/app/src/main/kotlin/org/vaccineimpact/api/app/controllers/AbstractController.kt

```diff
@@ -89,15 +89,28 @@ abstract class AbstractController(controllerContext: ControllerContext,
         return fullUrl
     }
 
-    fun objectCreation(context: ActionContext, urlFragment: String): String
+    fun objectCreation(
+            context: ActionContext,
+            urlFragment: String,
+            useControllerURL: Boolean = true
+    ): String
     {
-        val url = buildPublicUrl(urlFragment)
+        val url = buildPublicUrl(urlFragment, useControllerURL)
         context.addResponseHeader("Location", url)
         context.setResponseStatus(201)
         return url
     }
 
     fun okayResponse() = "OK"
 
-    fun buildPublicUrl(urlFragment: String) = Config["app.url"] + urlBase + urlComponent + urlFragment
+    fun buildPublicUrl(urlFragment: String, useControllerURL: Boolean = true): String
+    {
+        var url = Config["app.url"] + urlBase
```

Comment:
I prefer on the whole not to use boolean parameters to change function behaviour (2 separate, clearly named, functions preferable), and also I think this hides the url building logic away from where its used. Better to just have `objectCreation` take a fully qualified url rather than`urlFragment`, and then have a helper function to build the fully qualified urls, perhaps with varags so you can pass in as many paths as you want concatenated.

Also this url building logic should be under test!

Reply:
Good suggestions all. How about now?